File: algorithms/medium/number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int sum = accumulate(arr.begin(), arr.begin() + k, 0);
        int ct = (sum / k) >= threshold ? 1 : 0;
        for (int i = k; i < arr.size(); ++ i) {
          sum -= arr[i-k];
          sum += arr[i];
          if (sum / k >= threshold) ct++;
        }
        return ct;
    }
};

File: algorithms/medium/number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold.rs
impl Solution {
    pub fn num_of_subarrays(arr: Vec<i32>, k: i32, threshold: i32) -> i32 {
        let uk = k as usize;
        let mut sum: i32 = arr.iter().take(uk).sum::<i32>();
        let mut ct = 0;
        if sum / k >= threshold {
            ct += 1;
        }
        for i in uk..arr.len() {
            sum -= arr[i - uk];
            sum += arr[i];
            if sum / k >= threshold {
                ct += 1;
            }
        }
        ct
    }
}
